File: Project/Backend/src/ArtAuction.Infrastructure/Repositories/TagRepo.cs
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace ArtAuction.Infrastructure.Repositories;

public class TagRepo : ITagRepo
{
    // Attributes
    private readonly AppDbContext _dbContext;

    // Constructor
    public TagRepo(AppDbContext dbContext)
    {
        _dbContext = dbContext;
    }

    // Method
    public async Task<ICollection<Tag>> GetAllTags()
    {
        return await _dbContext.Tags.AsNoTracking().ToListAsync();
    }
}

File: ArtworkPostServices/src/main/java/com/artauction/artworkpostservices/Repositories/Interfaces/TagRepo.java
package com.artauction.artworkpostservices.Repositories.Interfaces;

import com.artauction.artworkpostservices.Entities.Tag;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TagRepo extends JpaRepository<Tag, Integer> {
}
